velox/dwio/dwrf/common/Compression.cpp

```diff
@@ -56,6 +59,33 @@ ZstdCompressor::compress(const void* src, void* dest, uint64_t length) {
   return ret;
 }
 
+#ifdef VELOX_ENABLE_QAT_ZSTD_OT
+class ZstdQatCompressor : public Compressor {
+ public:
+  explicit ZstdQatCompressor(int32_t level) : Compressor{level} {
+    zc = ZSTD_createCCtx();
+    QZSTD_startQatDevice();
+    sequenceProducerState = QZSTD_createSeqProdState();
+    ZSTD_registerSequenceProducer(
+        zc, sequenceProducerState, qatSequenceProducer);
+    ZSTD_CCtx_setParameter(zc, ZSTD_c_enableSeqProducerFallback, 1);
+  }
+
+  ZSTD_CCtx* zc;
+  void* sequenceProducerState;
+  uint64_t compress(const void* src, void* dest, uint64_t length) override;
+};
+
+uint64_t
+ZstdQatCompressor::compress(const void* src, void* dest, uint64_t length) {
+  size_t ret = ZSTD_compress2(zc, dest, length, src, length);
+  if ((int)ret <= 0) {
```

Comment:
wouldn't you need to do the zstd error checking, in addition to QAT to maintain compatiblity? similar to L52:56 in this file.

Reply:
> wouldn't you need to do the zstd error checking, in addition to QAT to maintain compatiblity? similar to L52:56 in this file.

line 82 is the error check.

Reply:
you're missing the check which skips size too small, from L54-55
if ((int)ret == ZSTD_ErrorCode::ZSTD_error_dstSize_tooSmall) {
      return length;

Reply:
added